package/gluon-core/luasrc/lib/gluon/upgrade/200-wireless

```diff
@@ -225,27 +197,30 @@ wireless.foreach_radio(uci, function(radio, index, config)
 	uci:delete('wireless', radio_name, 'basic_rate')
 
 	local band = radio.band
+	-- now we assign radios to roles
+	local roles = uci:get('gluon', 'band_' .. band, 'role')
+
 	if band == '2g' then
 		uci:set('wireless', radio_name, 'legacy_rates', false)
-		configure_mesh_wireless(radio, index, config)
-	elseif (band == '5g') then
-		-- ToDo: Remove in v2024.x
-		local hostapd_options = uci:get_list('wireless', radio_name, 'hostapd_options')
-		util.remove_from_set(hostapd_options, 'country3=0x4f')
-		uci:set_list('wireless', radio_name, 'hostapd_options', hostapd_options)
-
-		if is_outdoor() then
-			-- enforce outdoor channels by filtering the regdom for outdoor channels
-			uci:set('wireless', radio_name, 'country3', '0x4f')
-			configure_mesh_wireless(radio, index, config, true)
-		else
-			uci:delete('wireless', radio_name, 'country3')
+	end
+
+	if roles and util.contains(roles, 'mesh') then
+		if band == '2g' then
 			configure_mesh_wireless(radio, index, config)
+		elseif (band == '5g') then
+			if is_outdoor() then
+				-- enforce outdoor channels by filtering the regdom for outdoor channels
+				uci:set('wireless', radio_name, 'country3', '0x4f')
+			else
+				uci:delete('wireless', radio_name, 'country3')
+				configure_mesh_wireless(radio, index, config)
+			end
 		end
 	end
 
 	uci:set('wireless', radio_name, 'beacon_int', beacon_interval)
 
+	uci:set('wireless', "wan_" .. radio_name, 'disabled', util.contains(roles, 'private'))
```

Comment:
While this line is removed again in a later commit, the logic seems to be inverted.

Reply:
I deleted this line therefore, as it is not needed. Thanks!